### PR TITLE
APM-4831. Document the UAT option

### PR DESCRIPTION
**Pull request by johnmarston-nhs**

### CODE REVIEW

File: proxygen_cli/cli/command_main.py [hunk 1]
```diff
@@ -1,10 +1,7 @@
-import sys
-from typing import get_args
-from urllib import parse
 import click
 from proxygen_cli.lib.settings import SETTINGS
 
-from proxygen_cli.lib import constants, proxygen_api, settings, output, version
+from proxygen_cli.lib import proxygen_api, output
```
Comment (johnmarston-nhs) [inline proxygen_cli/cli/command_main.py:4]: Fettling up some unused imports.

File: proxygen_cli/cli/command_spec.py [hunk 2]
```diff
@@ -13,15 +14,26 @@
 
 @click.group()
 @click.option(
-    "--api", default=SETTINGS.api, help="Override the default API", show_default=True
+    "--api",
+    default=SETTINGS.api,
+    help="Override the default API",
+    show_default=True
+)
+@click.option(
```
Comment (johnmarston-nhs) [inline proxygen_cli/cli/command_spec.py:22]: Annoyingly, there's no way to add the `--uat` option to the root command and have it appear in the docs for both:
- `proxygen spec --help`
- `proxygen spec publish --help`
The only way around this is to add the `--uat` option to each sub-command (publish, get, delete) and then also to the root function.. and ignore the value.

File: proxygen_cli/test/command_spec_test.py [hunk 3]
```diff
@@ -110,6 +112,30 @@ def test_delete_spec_uat(patch_request, patch_access_token,
         assert path == "/apis/mock-api/spec/uat"
 
 
+@pytest.mark.parametrize("cmd, uat_expected", [
```
Comment (johnmarston-nhs) [inline proxygen_cli/test/command_spec_test.py:115]: The test just ensures that the `--uat` option appears in help messages for all relevant sub-commands (e.g. `proxygen spec publish`) **and** the root command `proxygen spec`.

**Review by harryhmc — COMMENTED** [review]
code is good, tests are good. only concern is how clear "Specification is in the UAT environment" is as a help message? i would expect a help message to describe what a flag does as opposed to an assumed context, e.g.
```--uat    Target the UAT instance of API catalogue```

**Comment by johnmarston-nhs** [comment]
> code is good, tests are good. only concern is how clear "Specification is in the UAT environment" is as a help message? i would expect a help message to describe what a flag does as opposed to an assumed context, e.g. `--uat Target the UAT instance of API catalogue`

Yeah,  think that's a fair point, and I like the way you've worded it.

File: proxygen_cli/cli/command_spec.py [hunk 4]
```diff
@@ -39,12 +41,12 @@ def spec(ctx, api, uat):
     "--no-confirm",
     is_flag=True,
     show_default=True,
-    help="Do not prompt for confirmation.",
+    help=HELP_NO_CONFIRM,
 )
 @click.option(
     "--uat",
```
Comment (harryhmc) [inline proxygen_cli/cli/command_spec.py:47]: would this be taking DRY too far?
```
UAT_KWARGS = {"default":False,"help": HELP_UAT, "is_flag": True}
[...]
@click.option("--uat", **UAT_KWARGS)
```

Comment (johnmarston-nhs) [reply proxygen_cli/cli/command_spec.py:56]: It looks good, yeah. I've nicked it. Also, applied the same to the "no confirm" stuff. I think it sets a good example - the APIM codebase under-uses constants IMO.

**Review by harryhmc — APPROVED** [review]

**Pull request merged**